cli/src/semgrep/core_runner.py

```diff
@@ -163,33 +164,51 @@ async def _core_stdout_processor(
         Updates progress bar one increment for every "." it sees from semgrep-core
         stdout
 
-        When it sees non-"." output it saves it to self._stdout
+        Increases the progress bar total for any number reported from semgrep-core
+        stdout
+
+        When it sees neither output it saves it to self._stdout
         """
         stdout_lines: List[bytes] = []
 
         # appease mypy. stream is only None if call to create_subproccess_exec
         # sets stdout/stderr stream to None
         assert stream
 
-        # Start out reading two bytes at a time (".\n")
-        bytes_to_read = 2
         while True:
             # blocking read if buffer doesnt contain any lines or EOF
-            line_bytes = await stream.read(n=bytes_to_read)
+            line_bytes = await stream.readline()
```

Comment:
I think we went from readlint to bytewise reading because we were filling up the pipe.

Reply:
Relevant: https://github.com/returntocorp/semgrep/issues/4693

I think we could keep track of where we are wrt total and pre-emptively breakout instead of trying to read the last line which is a giant json blob?

Reply:
Issue would be when `read` returns less than `n` bytes; per our conversation I'll just switch the termination condition to being based on the number of files scanned.